### ISSUE TITLE
Installing old version

### ISSUE THREAD

**Issue by pmatos** [greghendershott/travis-racket#34]
With `RACKET_VERSION == RELEASE` I expected **today** version 7.3. Instead got:
```
$ if [ ! -d $RACKET_DIR ]; then git clone https://github.com/greghendershott/travis-racket.git && cat travis-racket/install-racket.sh | bash;  fi
Cloning into 'travis-racket'...
Checking installer
Downloading https://pre-release.racket-lang.org/installers/racket-current-x86_64-linux.sh to ./racket-RELEASE.sh:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  2  117M    2 2674k    0     0  2938k      0  0:00:40 --:--:--  0:00:40 2935k
 33  117M   33 39.5M    0     0  20.6M      0  0:00:05  0:00:01  0:00:04 20.6M
 64  117M   64 75.1M    0     0  25.8M      0  0:00:04  0:00:02  0:00:02 25.7M
 94  117M   94  111M    0     0  28.4M      0  0:00:04  0:00:03  0:00:01 28.4M
100  117M  100  117M    0     0  28.7M      0  0:00:04  0:00:04 --:--:-- 28.7M
Making ./racket-RELEASE.sh executable:
Running ./racket-RELEASE.sh to install Racket:
This program will extract and install Racket v7.2.0.900.
```

**Comment by greghendershott** [comment]
Maybe `RELEASE` should have been named `PRE-RELEASE`. Furthermore it is simply whatever the core Racket devs decide to make available -- or not -- from time to time at https://pre-release.racket-lang.org/installers/.

The example .travis.yml has a comment:

```yaml
    # The RELEASE snapshot is only available during the release process.
```

1. Maybe this comment should be "stronger"?

2. Also, maybe the example .travis.yml should have the usage commented-out by default. It's simply not reliable as a "permanent" thing. I think the motivation of commit 84415da was, one could add this during "pre-release season", then remove it. Actually I never use it myself.

If you agree I'd be happy to make those changes.

**Comment by greghendershott** [comment]
p.s. To clarify, I wouldn't rename `RELEASE`. That would break existing `.travis.yml` files. (Well -- break them in a different way than they're arguably already broken.)  Instead, I mean two numbered items.

**Comment by greghendershott** [comment]
@pmatos I went ahead pushed a commit. If you have any comments please let me know. Otherwise I'll merge to `master` to close this issue

**Comment by greghendershott** [comment]
Also a heads-up to @gus-massa if you have any input?

**Comment by gus-massa** [comment]
The version v7.2.0.900 is essentially equal to v7.3, the only difference should be the banner.

Originally the idea was that the RELEASE version was available during the release cycle, but I asked nicely in the mailing list and they decide to keep the release version alive all the time (but frozen). 

I'm using it informally as an early warning system in a few projects. My idea to run constantly these project in HEAD, HEADCS and RELEASE, so when something get's broken I get an email from travis-ci. Most of the time, it is enough to use HEAD, HEAD, and the latest version(s).

But during the release cycle, it's useful to run it in RELEASE too. I'm too lazy to download the installer manually, and remember how to run all the correct test, so I prefer to run it in travis-ci automatically. It's an informal project, so some room for laziness is a feature :).

It's more easy to keep the test for RELEASE ruining all the time, instead of adding it when the release start and removing it when the release end and adding it when the release start and removing it when the release end, and forgetting the next time. I have 5 or 10 of these running, so it would be painful to enable and disable all of them each time.

In the main repository the [stable branch](https://github.com/racket/racket/commits/stable) tracks the latest published version, and is updated only once with every published version and hopefully must be always correct. Perhaps it is possible to create a STABLE target, with an official symlink in the racket server, or perhaps with a constant in travis-racket that is manually changed each time a new version is available. (I'd like STABLE, STABLE-1, STABLE-2, ... that track the latest published versions. I'm not sure if this is a good idea.)

In my opinion, the current configuration is correct, and by default RELEASE should be in the allow_failure section like now. (It may be commented by default, but I prefer that it is run so an enthusiastic user can spot a problem before the publication date.)

I think that it better to move the comment about RELEASE to a line between the last normal version and HEAD, HEADCS and RELEASE, and add a longer explanation about each versions, and why they can fail any day. (I tried to write something, but my versions are too bad.)

**Comment by pmatos** [comment]
If there are no more comments on this, I am happy for @greghendershott to close this.

**Comment by greghendershott** [comment]
@gus-massa Thank you for the explanation!

I tried to express this in commit d1bea53. Does that look OK?  If you have any suggestions please let me know. Otherwise I can squash commits and merge.

**Comment by greghendershott** [comment]
p.s. `STABLE` sounds good. But I think it would be a separate issue to tackle.

---

Also, as I think you allude, it would work better if implemented by the Racket core team --- as consistently named, immutable URLs. (The URLs never change. For some, just their target changes.)

(In fact, using that approach in general, would allow some of `install-racket.sh` to go away. A lot of it just handles URLs that have changed in unpredictable ways over the years. Some of `install-racket.sh` is almost a kind of localhost file. :smile:)

I've suggested this in the past. Also I think @jackfirth opened an issue about this, but I can't find it now.

However the core Racket team already has a lot on their plate, including but not limited to infrastructure/web/IT stuff. So I don't think it's reasonable to expect much on this front in the foreseeable future. And I'm not making a complaint, I'm fine with that, I'm just explaining.

**Comment by gus-massa** [comment]
I added some comments in https://github.com/gus-massa/travis-racket/commit/d17155e51a551fd51226df82c981be207195dfe9  (branch https://github.com/gus-massa/travis-racket/commits/19-6-E-greghendershott-issue-34 in case it is useful).

It's a rough attempt, so please feel free to improve or ignore them. I think it was easier to overwrite with "my" version instead of writing the comments here. (And sqhash it before merging.) In particular, I think that `indefinitely` is too long. Also, please remove/replace the `(hopefully)` I added.

**Comment by greghendershott** [comment]
@gus-massa Thanks! Although I'll probably do some prose editing, I understand your suggested changes, and I'll try to incorporate them.

**Comment by greghendershott** [comment]
Pushed commit e002ea8. Unless anyone has strong objections -- or notices a glaring typo! -- I'd like to go with this.

**Comment by gus-massa** [comment]
From the commit:

> that updates only **in** the weeks

I prefer 

> that updates only **during** the weeks

but I'm not a native English speaker, so I'm not sure which is better.


> As a result, you may leave this [the release version] in your .travis.yml all the time.

I'm not sure that it is necessary to highlight this, people will just assume that. But if you prefer to keep it, it's fine for me.

**Comment by greghendershott** [comment]
- I'll change "in" to "during" as you suggest. That's clearer.

- I'd like to keep this because it summarizes what you explained [above](https://github.com/greghendershott/travis-racket/issues/34#issuecomment-499729554): You made it possible for people to keep `RELEASE` in the matrix all the time -- and you want to encourage people to do so. Because I hadn't really understood and appreciated that, until now, I want to make sure other people do. Hopefully this extra sentence will help.

Thanks again!
